core/src/lib.rs

```diff
@@ -6,3 +10,58 @@ mod http;
 pub mod release;
 pub mod storage;
 pub mod tracker;
+
+/// The different games supported by the launcher
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Game {
+    Jumpy,
+    Punchy,
+}
+
+#[derive(thiserror::Error, Debug)]
+#[error("Invalid game ID")]
+pub struct InvalidGameId;
+
+impl FromStr for Game {
+    type Err = InvalidGameId;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "jumpy" => Ok(Game::Jumpy),
+            "punchy" => Ok(Game::Punchy),
+            _ => Err(InvalidGameId),
+        }
+    }
+}
+
+impl Game {
+    pub fn id(&self) -> &'static str {
+        match self {
+            Game::Jumpy => "jumpy",
+            Game::Punchy => "punchy",
+        }
+    }
+
+    pub fn name(&self) -> &'static str {
+        match self {
+            Game::Jumpy => "Jumpy",
+            Game::Punchy => "Punchy",
+        }
+    }
```

Comment:
Let's use `Display` implementation instead of having a `name` function. It feels more suitable that way.

https://doc.rust-lang.org/rust-by-example/hello/print/print_display.html